pydantic/types.py

```diff
@@ -687,6 +704,102 @@ def get_secret_value(self) -> bytes:
         return self._secret_value
 
 
+class PaymentCardBrand(Enum):
+    amex = 'American Express'
+    mastercard = 'Mastercard'
+    visa = 'Visa'
+    other = 'other'
+
+    def __str__(self) -> str:
+        return self.value
+
+
+class PaymentCardNumber(str):
+    """
+    Based on: https://en.wikipedia.org/wiki/Payment_card_number
+    """
+
+    strip_whitespace: ClassVar[bool] = True
+    min_length: ClassVar[int] = 12
+    max_length: ClassVar[int] = 19
+    bin: str
+    last4: str
+    brand: PaymentCardBrand
+
+    def __init__(self, card_number: str):
+        self.bin = card_number[:6]
+        self.last4 = card_number[-4:]
+        self.brand = self.get_brand(card_number)
+
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
```

Comment:
maybe this class could inherit from `ConstrainedStr`, they you can do

```py
yield from super().__get_validators__()
yield cls.validate_digits
...
```

Reply:
Okay, I'm easy either way. But in theory validators that aren't required shouldn't be called at all.

Reply:
Let's keep it the way it is then. I think it's much clearer what's happening the way it is now.